src/invidious/routes/api/v1/authenticated.cr

```diff
@@ -54,6 +54,60 @@ module Invidious::Routes::API::V1::Authenticated
     env.response.status_code = 204
   end
 
+  def self.get_history(env)
+    env.response.content_type = "application/json"
+    user = env.get("user").as(User)
+
+    page = env.params.query["page"]?.try &.to_i?
```

Comment:
You should either use `.to_u32` or `.clamp()` to make sure that you're not getting garbage (e.g: negative) page values. Ideally, an erroneous value (like `abcd`) should return an error 400 to the client.

Reply:
I didn't add the 400 since this pattern is reused across the code base, might make sense to refactor that separately (but also, I don't know if in doing so, that's going to break something)